### DISCUSSION TITLE
FieldArray - Access to keys on submit

### DISCUSSION THREAD

**Discussion by rraczdw** [react-hook-form/react-hook-form#2716] [Q&A] [answered]
Hey guys,

I have a complex form that supports create and edit with field arrays. I need to make a proper update on the server side and therefore I need the `id` of the items in the array that were passed to `useForm` as default values. 

For now I hacked it with a hidden input like:

```javascript
{paymentExpensesArray.fields.map((item, index) => (
    <div key={item.id}>
      <input
        type="hidden"
        name={`paymentExpenses[${index}].id`}
        ref={register()}
        defaultValue={item.id}
      />
     ....
```

This works but I am sure there is a better way. First time using RHF and really impressed!

Cheers,

Rob

**Comment by bluebill1049** [comment, marked as answer]
<img width="930" alt="Screen Shot 2020-08-26 at 10 43 58 am" src="https://user-images.githubusercontent.com/10513364/91241774-0d6cee80-e789-11ea-87fc-0409bc0515f2.png">

you can rename it.

**Reply by rraczdw** [reply]
Thanks but maybe my question was't stated well. My issue is that `id` is missing in onsubmit:

![image](https://user-images.githubusercontent.com/65724905/91312426-d9d69800-e779-11ea-977e-d1a6bbeed26e.png)

However I need that unchanged on the server. 

https://codesandbox.io/s/focused-saha-3vzo1?file=/src/App.js

Thanks!

**Reply by bluebill1049** [reply]
oh i see, put it into a hidden input `<input type="hidden" />`

**Reply by rraczdw** [reply]
Ya that's what I did, old school always works :D. I just thought it's quite common so there is less hacky way.
